aviary/mission/phase_builder_base.py

```diff
@@ -574,6 +419,127 @@ def _add_initial_guess_meta_data(cls, initial_guess: InitialGuess, desc=None):
         meta_data[name] = dict(
             apply_initial_guess=initial_guess.apply_initial_guess, desc=desc)
 
+    def set_time_options(self, user_options, targets=[]):
+        fix_initial = user_options.get_val('fix_initial')
+        duration_bounds = user_options.get_val('duration_bounds', units='s')
+        duration_ref = user_options.get_val('duration_ref', units='s')
+
+        self.phase.set_time_options(
+            fix_initial=fix_initial,
+            duration_bounds=duration_bounds,
+            units="s",
+            targets=targets,
+            duration_ref=duration_ref,
+        )
+
+    def add_TAS_state(self, user_options):
+        TAS_lower = user_options.get_val('TAS_lower', units='kn')
+        TAS_upper = user_options.get_val('TAS_upper', units='kn')
+        TAS_ref = user_options.get_val('TAS_ref', units='kn')
+        TAS_ref0 = user_options.get_val('TAS_ref0', units='kn')
+        TAS_defect_ref = user_options.get_val('TAS_defect_ref', units='kn')
+        self.phase.add_state(
+            "TAS",
+            fix_initial=user_options.get_val('fix_initial'),
+            fix_final=False,
+            lower=TAS_lower,
+            upper=TAS_upper,
+            units="kn",
+            rate_source="TAS_rate",
+            targets="TAS",
+            ref=TAS_ref,
+            ref0=TAS_ref0,
+            defect_ref=TAS_defect_ref,
+        )
```

Comment:
actually, since these all have the same suffixes, these could all be combined into a single `add_state` that takes state_name, rate_name, and units as arguments

Reply:
Thanks Carl, good considerations! Will tack on to another branch where I've got more changes already.